modules/rostests/kmtests/ntos_io/IoDeviceInterface.c

```diff
@@ -1,125 +1,214 @@
-        SymbolicLinkListPtr += wcslen(SymbolicLinkListPtr) + 1)
-   {
-      UNICODE_STRING SymbolicLink;
-      UNICODE_STRING AliasSymbolicLink;
-
-      SymbolicLink.Buffer = SymbolicLinkListPtr;
-      SymbolicLink.Length = SymbolicLink.MaximumLength = wcslen(SymbolicLinkListPtr);
-      RtlInitUnicodeString(&AliasSymbolicLink, NULL);
-      IoGetDeviceInterfaceAlias(
-         &SymbolicLink,
-         &AliasGuid,
-         &AliasSymbolicLink);
-      if (AliasSymbolicLink.Buffer != NULL)
-      {
-         DPRINT("[PnP Test] Original: %S\n", SymbolicLinkListPtr);
-         DPRINT("[PnP Test] Alias: %S\n", AliasSymbolicLink.Buffer);
-      }
-   }
-#endif
-
-   ExFreePool(SymbolicLinkList);
+    RtlInitUnicodeString(&SymbolicLinkName, SymbolicLink);
+    RtlInitUnicodeString(&KeyName, L"ReactOS_kmtest");
+
+    for (size_t n = 0; n < RTL_NUMBER_OF(Types); ++n)
+    {
+        HANDLE DeviceInterfaceKey, DeviceInterfaceSubKey;
+        OBJECT_ATTRIBUTES ObjectAttributes;
+
+        NTSTATUS Status = IoOpenDeviceInterfaceRegistryKey(&SymbolicLinkName, KEY_CREATE_SUB_KEY, &DeviceInterfaceKey);
+
+        if (NT_SUCCESS(Status))
+        {
+            trace("IoOpenDeviceInterfaceRegistryKey(): success: %d %p\n", n, DeviceInterfaceKey);
+        }
+        else
+        {
+            trace("IoOpenDeviceInterfaceRegistryKey(): fail: %d 0x%x\n", n, Status);
+        }
+
+        InitializeObjectAttributes(&ObjectAttributes,
+                                   &KeyName,
+                                   OBJ_CASE_INSENSITIVE | OBJ_KERNEL_HANDLE,
+                                   DeviceInterfaceKey,
+                                   NULL);
+        Status = ZwCreateKey(&DeviceInterfaceSubKey,
+                             KEY_WRITE,
+                             &ObjectAttributes,
+                             0,
+                             NULL,
+                             REG_OPTION_NON_VOLATILE,
+                             NULL);
+
+        if (NT_SUCCESS(Status))
+        {
+            trace("ZwCreateKey(): successfully created subkey: %d %p\n", n, DeviceInterfaceSubKey);
+
+            ZwDeleteKey(DeviceInterfaceSubKey);
+            ZwClose(DeviceInterfaceSubKey);
+        }
+        else
+        {
+            trace("ZwCreateKey(): failed to create a subkey: %d 0x%x\n", n, Status);
+        }
+
+        ZwClose(DeviceInterfaceKey);
+    }
 }
 
 static
 VOID
-Test_IoRegisterDeviceInterface(VOID)
+Test_IoGetDeviceInterfaceAlias(PWSTR SymbolicLink)
 {
-    GUID Guid = {0x378de44c, 0x56ef, 0x11d1, {0xbc, 0x8c, 0x00, 0xa0, 0xc9, 0x14, 0x05, 0xdd}};
-    DEVICE_OBJECT DeviceObject;
-    EXTENDED_DEVOBJ_EXTENSION DeviceObjectExtension;
-    DEVICE_NODE DeviceNode;
     UNICODE_STRING SymbolicLinkName;
-    NTSTATUS Status;
 
-    RtlInitUnicodeString(&SymbolicLinkName, L"");
+    RtlInitUnicodeString(&SymbolicLinkName, SymbolicLink);
+
+    for (size_t n = 0; n < RTL_NUMBER_OF(Types); ++n)
```

Comment:
Same here, move declaration of n upwards please.

Reply:
Done.